pkg/restore/restore.go

```diff
@@ -1249,7 +1249,21 @@ func (ctx *restoreContext) restoreItem(obj *unstructured.Unstructured, groupReso
 		errs.Add(namespace, err)
 		return warnings, errs
 	}
-	if isAlreadyExistsError {
+	// check if we want to treat the error as a warning, in some cases the creation call might not get executed due to object API validations
+	// and Velero might not get the already exists error type but in reality the object already exists
+	objectExists := false
+	if !isAlreadyExistsError {
+		_, err := resourceClient.Get(obj.GetName(), metav1.GetOptions{})
+		if !apierrors.IsNotFound(err) {
+			objectExists = true
+		}
+		if err != nil && !apierrors.IsNotFound(err) {
+			errs.Add(namespace, err)
+			return warnings, errs
+		}
```

Comment:
This reduce call to !apierrors.IsNotFound(err). More performance?
```suggestion
		if !apierrors.IsNotFound(err) {
			objectExists = true
		}
		if err != nil && objectExists {
			errs.Add(namespace, err)
			return warnings, errs
		}
```